### DISCUSSION TITLE
[STUDY GROUP]: GitHub Actions Exam

### DISCUSSION THREAD

**Discussion by v-fidelusaleksander** [v-fidelusaleksander/ghcertified#165] [Study Together!]
# GitHub Actions :book: STUDY GROUP

Hello everyone,

Many of us are preparing for the GitHub Certification Exams and I thought it would be beneficial to start a discussion where we can share:

-  experiances
-  study tips
-  ask questions 
- anything else regarding any of the topics covered in the exam 

Looking forward to hearing from you all! :rocket:

**Comment by reubendevries** [comment]
Does anyone feel like the answer to this question is incorrect?

>Q13: A workflow was initially run on commit A and failed. You fixed the workflow with the subsequent commit B. When you re-run that workflow it will run with code from which commit?

For some reason it says it will reuse commit A but I don't think that's true. I think once you make a new commit, it won't use commit A to build the workflow but commit B, and you wouldn't want it to use commit A, because commit A was failing - am I taking crazy pills here?

I think the documentation they are referring to is saying something different, at no point in the documentation does it mention making a change and then committing and pushing that change. Granted it doesn't mention making a push in the question, so either it's a poorly worded question that is either testing your reading comprehension (Which is  a pet peeve of mine when technical tests do that or it's just flat out wrong).

**Reply by marcosboger** [reply]
A bit late to it, but I believe the keyword here is **re-run**. If you manually trigger or run the workflow again, sure, it will use the latest commit. However, when **re-running**, it will use the workflow from the original commit. 

Created a simple example here in this repo, feel free to check it out: https://github.com/marcosboger/workflow-rerun-test

I've ran the original workflow and it had the following output:
![image](https://github.com/user-attachments/assets/4dd42e45-5558-4345-960a-c5b6e1a812e6)
![image](https://github.com/user-attachments/assets/bed1ffd1-e3ee-4364-a205-f2b80f3de48f)



I've commited changing the hello world message:
![image](https://github.com/user-attachments/assets/a2439d99-31d8-497b-b357-ee3d31dc279b)
 
If I **re-run**, it still prints the old hello world message:
![image](https://github.com/user-attachments/assets/f96891fa-8b94-4e1b-9cdf-b190cc9ca2e5)
![image](https://github.com/user-attachments/assets/b5b9e45b-d589-4dd4-abe5-171cd79d2dff)

If I trigger manually a new run, it prints the new message:
![image](https://github.com/user-attachments/assets/c20e04c0-71ce-4420-986f-c66623805064)

**Reply by reubendevries** [reply]
Makes perfect sense, thanks for clarifying!

**Comment by ThierryKassa** [comment]
Hi everyone. I have my exam tomorrow and I need help on this question. My main concern is "include" allow job duplication in the matrix? I asked ChatGPT and it says the answer is **6** with this comment: “The include keyword lets you add additional combinations to the matrix. **If the parameters match an existing combination, the job is duplicated.**”

Q7: Given the following configuration, how many jobs will GitHub Actions run when this matrix is evaluated?
strategy:
  matrix:
    os: [ubuntu-latest, windows-latest]
    node: [14, 16]
    include:
      - os: macos-latest
        node: 18
      - os: ubuntu-latest
        node: 14


7 jobs

No jobs will run because the syntax is invalid.

5 jobs

6 jobs

4 jobs

**Reply by ThierryKassa** [reply]
I tested in my repo and outputs 5 jobs

<img width="1412" height="826" alt="mat1" src="https://github.com/user-attachments/assets/75acf499-eb43-49b2-8bea-94b4eb864d46" />

<img width="1866" height="733" alt="mat2" src="https://github.com/user-attachments/assets/3035ddfa-59a4-441e-a69f-4c1319f400ff" />

**Reply by v-fidelusaleksander** [reply]
That's the best way to check things - test on your own repo - makes you remember so much more 👍 Good job


Hope you passed the exam @ThierryKassa ! 🚀
